src/site/docs/dart-up-and-running/ch05.markdown

```diff
@@ -479,34 +477,27 @@ objects. Here's an example from the GitHub `search()` method:
 {% prettify dart %}
 import 'package:http/http.dart' as http_client;
 // ...
-Stream<SearchResult> search(String input) {
-  var query = {
-    'q': 'language:dart $input'
-  };
-  var searchUri = new Uri.https(
-      'api.github.com',
-      '/search/repositories',
-      query);
-  var controller = new StreamController();
-  http_client.get(searchUri)
-    .then((http_client.Response response) {
-      if (response.statusCode != HttpStatus.OK) {
-        throw "Bad status code: ${response.statusCode}, "
-              "message: ${response.reasonPhrase}";
-      }
-      var json = JSON.decode(response.body);
-      json.putIfAbsent('items', () => []);
-      json['items']
-        .expand(/* Delete items with empty descriptions. */)
-        .take(3)
-        .forEach((item) {
-          controller.add(new SearchResult(
-              item['description'], item['html_url']));
-        });
-    })
-    .catchError(controller.addError)
-    .whenComplete(controller.close);
-  return controller.stream;
+Stream<SearchResult> search(String input) async* {
```

Comment:
We should add comments before each {% prettify ...%} section saying which file the code is from. For example:

```
{% comment %}
lib/src/github_search_engine.dart
{% endcomment %}
{% prettify dart %}
...
```